ratis-server/src/main/java/org/apache/ratis/server/impl/LeaderElection.java

```diff
@@ -234,12 +234,17 @@ LifeCycle.State getCurrentState() {
 
   @Override
   public void run() {
+    if (server.initializing()) {
+        LOG.info("{}: skip running since the server is initializing", this);
+        return;
+    }
     if (!lifeCycle.compareAndTransition(STARTING, RUNNING)) {
       final LifeCycle.State state = lifeCycle.getCurrentState();
       LOG.info("{}: skip running since this is already {}", this, state);
       return;
     }
 
+
```

Comment:
Please revert this whitespace change.

Reply:
Sorry, I deleted this blank line, but forgot to push it again.

The implementation you provided is very good, I will modify it right away